operator/src/main/java/oracle/kubernetes/operator/helpers/PodStepContext.java

```diff
@@ -306,10 +307,35 @@ private void addContainerPort(List<V1ContainerPort> ports, NetworkAccessPoint na
   private void addContainerPort(List<V1ContainerPort> ports, String name,
                                 @Nullable Integer listenPort, String protocol) {
     if (listenPort != null) {
+      name = createContainerPortName(ports, name);
       ports.add(new V1ContainerPort().name(name).containerPort(listenPort).protocol(protocol));
     }
   }
 
+  private String createContainerPortName(List<V1ContainerPort> ports, String name) {
+    //Container port names can be a maximum of 15 characters in length
+    if (name.length() > LegalNames.LEGAL_CONTAINER_PORT_NAME_MAX_LENGTH) {
+      String portNamePrefix = getPortNamePrefix(name);
+      // Find ports with the name having the same first 12 characters
+      List<V1ContainerPort> containerPortsWithSamePrefix = ports.stream().filter(port ->
+              portNamePrefix.equals(getPortNamePrefix(port.getName()))).collect(Collectors.toList());
+      int index = containerPortsWithSamePrefix.size() + 1;
+      String indexStr = String.valueOf(index);
+      // zero fill to the left for single digit index (e.g. 01)
+      if (index < 10) {
```

Comment:
I guess we're hoping that no one ever has 100 channels :)

Reply:
All kidding aside... It is extremely unlikely that there will be over 100 channels that are also longer than the maximum name, but we ought to do something here like generate a specific error message.

Reply:
:) In addition to 100 channels having longer than the maximum name, these 100 channels also need to have the same first 12 characters. Anyways, I'll look into adding a specific error message.

Reply:
Added an error message when more than 100 channels are configured with long names in f9d576a .